simpleParser.go

```diff
@@ -0,0 +1,470 @@
+package gojsonsm
+
+import (
+	//	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var ErrorNotFound error = fmt.Errorf("Error: Specified resource was not found")
+var ErrorNoMoreTokens error = fmt.Errorf("Error: No more token found")
+var ErrorNeedToStartOneNewCtx error = fmt.Errorf("Error: Need to spawn one subcontext")
+var ErrorNeedToStartNewCtx error = fmt.Errorf("Error: Need to spawn subcontext")
+var ErrorParenMismatch error = fmt.Errorf("Error: Parenthesis mismatch")
+var NonErrorOneLayerDone error = fmt.Errorf("One layer has finished")
```

Comment:
it is strange to return an error indicating there is no error. may be cleaner to return (status, error). so that oneLayerFinished can be indicated by status rather than error.